pkg/bench/summary.go

```diff
@@ -0,0 +1,109 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Tetragon
+
+package bench
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	"time"
+
+	"github.com/cilium/tetragon/pkg/metrics/ringbufmetrics"
+	"github.com/fatih/color"
+	"github.com/prometheus/client_golang/prometheus"
+	dto "github.com/prometheus/client_model/go"
+)
+
+// Summary gathers benchmark results. Serializes to JSON.
+// This is updated from multiple places concurrently, but currently
+// there is no overlap on writes, so this isn't yet protected by a mutex.
+type Summary struct {
+	Args *Arguments
+
+	ExitEvents, ExecEvents int64
+
+	StartTime          time.Time
+	EndTime            time.Time
+	RunTime            time.Time
+	SetupDurationNanos time.Duration
+	TestDurationNanos  time.Duration
+
+	JSONEncodingDurationNanos time.Duration
+	ExportStats               CountingDiscardWriter
+
+	TetragonCPUUsage CPUUsage
+
+	BpfStats map[int64]*BpfProgStats
```

Comment:
Is CPU id the key here?

Reply:
Is see. Could you please add a comment about what the key is?

Reply:
yep, will do